src/wasm-binary.h

```diff
@@ -1011,7 +1042,8 @@ class WasmBinaryBuilder {
 
   void readMemory() {
     if (debug) std::cerr << "== readMemory" << std::endl;
-    verifyInt8(BinaryConsts::Memory);
+    if (wasm.memory.read) abort();
```

Comment:
is this check and the checks below verifying that each section is seen once?

i would prefer a more generic approach to this instead of adding a field to `wasm.memory`, checking sizes, etc. How about an `std::set` of the seen sections, used in the main section `for` loop?